packages/client/src/net/protocol/ethprotocol.ts

```diff
@@ -219,18 +224,20 @@ export class EthProtocol extends Protocol {
       encode: ({ reqId, txs }: { reqId: bigint; txs: TypedTransaction[] }) => {
         const serializedTxs = []
         for (const tx of txs) {
-          switch (tx.type) {
-            case 0:
-              serializedTxs.push(tx.raw())
-              break
-            case 3:
-              serializedTxs.push((tx as BlobEIP4844Transaction).serializeNetworkWrapper())
-              break
-            default:
-              serializedTxs.push(tx.serialize())
-              break
+          if (isLegacyTx(tx)) {
+            serializedTxs.push(tx.raw())
+            break
+          }
+
+          if (isBlobEIP4844Tx(tx)) {
+            serializedTxs.push(tx.serializeNetworkWrapper())
+            break
```

Comment:
break here will break the for loop, continue is better , but overall it should be chained in if else if...

Reply:
updated

Reply:
Thanks!